client/plrdlg.cpp

```diff
@@ -750,14 +757,27 @@ void plr_report::call_meeting()
 }
 
 /**
-   Slot for canceling threaty
+   Slot for canceling treaty
  */
 void plr_report::plr_cancel_threaty()
 {
   dsend_packet_diplomacy_cancel_pact(
       &client.conn, player_number(other_player), CLAUSE_CEASEFIRE);
 }
 
+/**
+   Slot for diplomacy
+ */
+void plr_report::plr_diplomacy()
+{
+  int i;
+  i = queen()->gimmeIndexOf(QStringLiteral("DDI"));
```

Comment:
Remark for the future: `int i = queen()->gimmeIndexOf(...);` in a single line and simplify if clauses when possible.